tests/rptest/tests/cluster_features_test.py

```diff
@@ -423,6 +423,8 @@ def test_old_node_join(self):
 
         # Restart it with a sufficiently recent version and join should succeed
         self.installer.install([old_node], RedpandaInstaller.HEAD)
+        self.redpanda.stop_node(old_node)
+        self.redpanda.clean_node(old_node, preserve_current_install=True)
```

Comment:
So, without these lines, I think what's happening is this:

- The test downgrades `old_node`.
- Since `old_node` was previously cleaned (controller and kvstore logs wiped out?), there's nothing to replay into the in-memory `kvstore`, so we never save a local feature snapshot. Or something.
- Then we reinstall HEAD. At startup, we have a controller log from the previous run but no FT snapshot, so the assertion triggers.

I guess the question is whether that can occur in normal operation or it's an isolated side effect of the test. If the former, then the assertion probably isn't expressive enough. This feels extremely subtle to me, but it could be my own lack of familiarity.